students/joejohnsto/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Oct 25 14:42:34 2019
+
+@author: jjohnston
+"""
+
+
+# Task One
+
+t = (2, 123.4567, 10000, 12345.67)
+s = 'file_{:0=3d} :  {:.2f}, {:.2e}, {:.2e}'
+
+s.format(*t)
+
+
+# Task Two
+# redo task one using f-string
+s = f'file_{t[0]:03} :  {t[1]:.2f}, {t[2]:.2e}, {t[3]:.2e}'
+
+
+# Task Three
+
+t = (1,2,3,4)
+
+def formatter(t: tuple):
+    """Return a string describing the input tuple"""
+    s = 'The {} numbers are: ' + '{}, '*(len(t)-1) + '{}'
+    return s.format(len(t),*t)
+
+
+# Task Four
+    
+t = ( 4, 30, 2017, 2, 27)
+s = f'{t[3]:02} {t[4]} {t[2]} {t[0]:02} {t[1]}'
+
+
+# Task Five
+
+l = ['oranges', 1.3, 'lemons', 1.1]
+s = f'The weight of an {l[0][:-1]} is {l[1]} and the weight of a {l[2][:-1]} is {l[3]}'
+s2 = f'The weight of an {l[0][:-1].upper()} is {l[1]*1.2} and the weight of a {l[2][:-1].upper()} is {l[3]*1.2}'
+
+
```

Comment:
Where did you actually PRINT for tasks 1-5?

Reply:
Because of that, I deduced you 1 point, but you have 1 extra...

Reply:
I didn't actually print the resultant strings for Task 1-5, but the assignment description does not explicitly state that we needed to print, just that we needed to write code that would build the string correctly.

Reply:
I acknowledge this and I am returning back your 1 point. Thanks.